src/jailer/src/cgroup.rs

```diff
@@ -314,18 +365,26 @@ impl CgroupV1 {
 }
 
 impl Cgroup for CgroupV1 {
-    fn write_value(&self) -> Result<(), JailerError> {
+    fn add_property(&mut self, file: String, value: String) -> Result<(), JailerError> {
+        self.base.properties.push(CgroupProperty { file, value });
+        Ok(())
+    }
+
+    fn write_values(&self) -> Result<(), JailerError> {
         let location = &mut self.base.location.clone();
 
         // Create the cgroup directory for the controller.
         fs::create_dir_all(&self.base.location)
             .map_err(|err| JailerError::CreateDir(self.base.location.clone(), err))?;
 
-        // Write the corresponding cgroup value. inherit_from_parent is used to
-        // correctly propagate the value if not defined.
-        inherit_from_parent(location, &self.base.file, self.cg_parent_depth)?;
-        location.push(&self.base.file);
-        writeln_special(location, &self.base.value)?;
+        for property in self.base.properties.iter() {
+            let file_location = &mut location.clone();
+            // Write the corresponding cgroup value. inherit_from_parent is used to
+            // correctly propagate the value if not defined.
+            inherit_from_parent(location, &property.file, self.cg_parent_depth)?;
+            file_location.push(&property.file);
+            writeln_special(file_location, &property.value)?;
+        }
```

Comment:
I tried to keep the amount of unrelated changes to a minimum because it's already a pretty big refactor and I didn't want to change the existing logic. I can push a followup with these changes.

Reply:
If its in a separate commit, then including it here is perfectly fine :)

Reply:
will do then! thx